Error message when the Fortran compiler needs -fallow-argument-mismatch

https://github.com/pmodels/mpich/blob/9d9d208011b7714daf9ef680f015bb18ff3aa5b6/configure.ac#L1995

It might be helpful to replace `Rerun configure with FCFLAGS=$addarg` with `Rerun configure with FFLAGS=$addarg and FCFLAGS=$addarg`

I was compiling the last release of mpich-4.0 with gcc-11.2.0 :
```
../mpich-4.0/configure FCFLAGS=-fallow-argument-mismatch --prefix=/home/cflage01/opt/mpich/mpich-4.0_bld --enable-debuginfo --with-device=ch4:ofi --enable-g=all
```
This leads to
```
checking whether gfortran allows mismatched arguments... yes, with -fallow-argument-mismatch
configure: error: The Fortran compiler gfortran does not accept programs that call the same routine with arguments of different types without the option -fallow-argument-mismatch.  Rerun configure with FFLAGS=-fallow-argument-mismatch
configure: error: ../../../mpich-4.0/test/mpi/configure failed for test/mpi
```
Using
```
../mpich-4.0/configure FFLAGS=-fallow-argument-mismatch FCFLAGS=-fallow-argument-mismatch --prefix=/home/cflage01/opt/mpich/mpich-4.0_bld --enable-debuginfo --with-device=ch4:ofi --enable-g=all
```
does not produce any error

--- a/configure.ac
+++ b/configure.ac
@@ -1992,7 +1992,7 @@ if test "$enable_f77" = "yes" ; then
 	# that turns off checking for *everything* is not something that
 	# configure should do - if the user wants this, they can follow
 	# the instructions in the following error message.
-        AC_MSG_ERROR([The Fortran compiler $FC does not accept programs that call the same routine with arguments of different types without the option $addarg.  Rerun configure with FCFLAGS=$addarg])
+        AC_MSG_ERROR([The Fortran compiler $FC does not accept programs that call the same routine with arguments of different types without the option $addarg.  Rerun configure with FFLAGS=$addarg and FCFLAGS=$addarg])
     fi
 
     bindings="$bindings f77"

